receiver/splunkhecreceiver/splunkhec_to_metricdata.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package splunkhecreceiver
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/common/splunk"
+)
+
+// SplunkHecToMetricsData converts Splunk HEC metric points to
+// consumerdata.MetricsData. Returning the converted data and the number of
+// dropped time series.
+func SplunkHecToMetricsData(logger *zap.Logger, events []*splunk.Event, resourceCustomizer func(pdata.Resource)) (pdata.Metrics, int) {
+
+	// TODO: not optimized at all, basically regenerating everything for each
+	// 	data point.
+	numDroppedTimeSeries := 0
+	md := pdata.NewMetrics()
+
+	for _, event := range events {
+		resourceMetrics := pdata.NewResourceMetrics()
+		resourceMetrics.InitEmpty()
+
+		metrics := pdata.NewInstrumentationLibraryMetrics()
+		metrics.InitEmpty()
+
+		labelKeys, labelValues := buildLabelKeysAndValues(event.Fields)
+		if len(labelKeys) > 0 {
+			resourceMetrics.Resource().InitEmpty()
+		}
+		for i, k := range labelKeys {
+			resourceMetrics.Resource().Attributes().InsertString(k, labelValues[i])
```

Comment:
This should be the same host/source thing as with logs.  Or are metrics different?

Reply:
Fixed, I missed it went under data points.